### PR TITLE
Improved editor docs

### LINKED ISSUE #10171: Editor Help duplicated and no line breaks
**Issue description:**

sometimes the editor help is buggy.
Here is a screenshot:
screen shot 2017-08-08 at 19 00 36

I can't tell when it happens. 

**Issues are:**
 - missing newLines, indents
 - duplicate entries of property and method description list

### It is formatted like this:

WORKING inheritance/class section
WORKING property list
WORKING method list
WORKING description
BROKEN property description list
BROKEN method description list
WORKING property description list
WORKING method description list

BROKEN: no tabs, no indents, no newline

### PR DESCRIPTION
**Pull request by djrm**
Fixes a lot of problems with the editor doc (maybe all of them).
closes #10171

visual improvements
screenshot from 2017-09-13 18-30-07

screenshot from 2017-09-13 20-06-06

also added a handy docs index, to easily navigate across docs sections.
screenshot from 2017-09-13 19-55-03

changed the code font to Mononoki, looks better in low res screens, and seems to be a more fitting font.

### CODE REVIEW

**Comment by toger5** [comment]
Nice really cool you fixed  #10171

As far as I can tell from your code you changed it so that the docs use the editor theme colors not the code theme colors.

Did you not like the same look for script and help?

**Comment by djrm** [comment]
@toger5 no, i didnt like that, its better to be free to make the help look, without worring about the user script editor theme, it also has more advantages, like support for white theme, and it is clearer to identify what is more important, i think it looks really good now.

**Comment by djrm** [comment]
@toger5 also if at some point we decide to highlight the codeblocks, using the using the script editor colors might look really strange and confusing.

**Comment by toger5** [comment]
@djrm really good points!
You are right it looks really great now. and you are right its a big advantage to have it look really well with the rest of the editor.

Just also liked the unifyed style for script + help.

**Comment by reduz** [comment]
whoa, you rock

File: editor/editor_help.cpp [hunk 1]
```diff
@@ -1419,70 +1454,16 @@ static void _add_text_to_rt(const String &p_bbcode, RichTextLabel *p_rt) {
...
-				}
-			}
-		}
-	}
-
-	//change newlines to double newlines
-
-	CharType dnls[2] = { 0xFFFF, 0 };
-	bbcode = bbcode.replace(dnls, "\n");
-
```
Comment (toger5) [inline editor/editor_help.cpp]: @djrm is this double space section the part that caused the error with duplicating the docs in one line?

sometimes I also had selection issues. I guess they are also gone now?

Comment (djrm) [reply editor/editor_help.cpp]: @toger5 there were a lot issues, removing all that in that function only solved the codeblock thing, but the overlapping text was caused by a wrong pop call (was hard to find). also toger there were never duplication, if you check your screenshot you will see that it appears to be so, but it is just the way members descriptions are added, it seems that all the methods you saw used the same Setter and Getter, thats why it looked duplicated.

Comment (toger5) [reply editor/editor_help.cpp]: I'm pretty sure there were duplicates too.

But yea I also spend time trying to find stuff in the editor_help class but it's super hard (kudos for you !!)

hera are some more screenshots (sorry they take a lot of space...) that show the dulicated sections.
screen shot 2017-09-14 at 04 39 57

File: thirdparty/fonts/LICENSE_Mononoki [hunk 2]
```diff
@@ -1,8 +1,9 @@
-Copyright 2010, 2012 Adobe Systems Incorporated (http://www.adobe.com/), with Reserved Font Name 'Source'. All Rights Reserved. Source is a trademark of Adobe Systems Incorporated in the United States and/or other countries.
```
Comment (akien-mga) [inline thirdparty/fonts/LICENSE_Mononoki]: Please update `thirdparty/README.md` and `COPYRIGHT.txt` too.

**Comment by akien-mga** [comment]
Can we have a before/after of the script editor with the changed font?

File: thirdparty/fonts/LICENSE_Mononoki (hunk 2 above)
Comment (djrm) [reply thirdparty/fonts/LICENSE_Mononoki]: could not find anything in README.md

**Comment by n-pigeon** [comment]
Woooo nice. I want that font in script editor! :D

**Comment by djrm** [comment]
@akien-mga Might not be a fair comparison but here it is
screenshot from 2017-09-14 01-00-33

**Comment by djrm** [comment]
@n-pigeon it is there too

File: thirdparty/fonts/LICENSE_Mononoki (hunk 2 above)
Comment (akien-mga) [reply thirdparty/fonts/LICENSE_Mononoki]: https://github.com/godotengine/godot/blob/master/thirdparty/README.md#fonts

Comment (djrm) [reply thirdparty/fonts/LICENSE_Mononoki]: damn, should be fixed now.

**Comment by toger5** [comment]
@djrm just as an idea.. do we want the default code editor color theme (not the docs theme) also to be using generated colors? so that default creates good results with light and dark theme?

also I think the code editor colors could benefit from some fresh wind?

**Comment by toger5** [comment]
Using self made themes would still be an option but it would not guarantee good support for light and dark

**Comment by djrm** [comment]
@toger5 auto generated script theme would be wonderful, but of course should be an option, and i think it would support light themes and dark themes by default.

**Comment by toger5** [comment]
@djrm It would be easy to make it an option since it could just be the default code theme.
so instead of loading the default as hardcoded values from c++ it uses the theme color constants and a little math. AND the accent color

**Comment by djrm** [comment]
@toger5 im seriously thinking about this, i might implement it.

**Pull request merged**